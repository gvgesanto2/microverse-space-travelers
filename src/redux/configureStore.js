import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import rocketReducer from './rocket/rocket.reducer';
import getRocketsFromApi from './rocket/rocketApiAction';

// Initial Data
const preloadedState = {
  rockets: [],
};

// Establish the root reducer
const rootReducer = combineReducers({
  rockets: rocketReducer,
});

// Set up the Store
const ulisesStore = configureStore({
  reducer: rootReducer,
  preloadedState,
});

ulisesStore.dispatch(getRocketsFromApi());

export default ulisesStore;
