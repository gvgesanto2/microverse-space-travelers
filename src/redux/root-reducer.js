/* eslint-disable import/prefer-default-export */
import { combineReducers } from 'redux';
import missionReducer from './mission/mission.reducer';
import rocketReducer from './rocket/rocket.reducer';

export const rootReducer = combineReducers({
  rocket: rocketReducer,
  mission: missionReducer,
});
