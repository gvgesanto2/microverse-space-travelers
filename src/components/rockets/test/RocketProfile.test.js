import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../redux/store';
import MyProfile from '../../../pages/my-profile/my-profile.component';

const renderRocketProfile = () => {
  const container = render(
    <Provider store={store}>
      <MyProfile />
    </Provider>,
  );
  return container;
};

describe('My Profile page', () => {
  it('Renders my profile component correctly', () => {
    expect(renderRocketProfile()).toMatchSnapshot();
  });
});
