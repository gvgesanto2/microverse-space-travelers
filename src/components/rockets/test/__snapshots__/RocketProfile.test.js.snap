// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Profile page Renders my profile component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="my-profile"
      >
        <div
          class="missions-reserved"
        >
          <h2
            class="missions-reserved__title"
          >
            My Missions
          </h2>
          <ul
            class="missions-reserved__list"
          />
        </div>
        <section
          class="rocket-profile"
        >
          <h2>
            My Rockets
          </h2>
          <p>
            You have not reserved any rockets
          </p>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="my-profile"
    >
      <div
        class="missions-reserved"
      >
        <h2
          class="missions-reserved__title"
        >
          My Missions
        </h2>
        <ul
          class="missions-reserved__list"
        />
      </div>
      <section
        class="rocket-profile"
      >
        <h2>
          My Rockets
        </h2>
        <p>
          You have not reserved any rockets
        </p>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
